### ISSUE TITLE
Cannot install COCO Annotator using Docker either on Linux or Windows

### ISSUE THREAD

**Issue by FeriBolour** [jsbroks/coco-annotator#517]
Hello Everyone,
I haven't worked with Docker before trying to use this tool. I followed all the installation steps from docker and COCO-Annotator both on Linux and Windows and it always gets stuck in the middle of the installation at the same place.

When I run `$ docker-compose up` I get this error:
`.`
`.`
`.`
`annotator_workers    | [2021-10-12 00:17:02,694: INFO/MainProcess] mingle: all alone`
`annotator_workers    | [2021-10-12 00:17:02,807: INFO/MainProcess] celery@67c1eba9b51e ready.`
`annotator_message_q  | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> closing AMQP connection <0.778.0> (172.18.0.4:59266 -> 172.18.0.2:5672):`
`annotator_message_q  | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> missed heartbeats from client, timeout: 60s
`

I would appreciate any kind of help. Am I doing something wrong?

Thank You

**Comment by tejasri19** [comment]
Verify docker and docker-compose versions.

**Comment by pinglin-zhang** [comment]
> Verify docker and docker-compose versions.
for me:
Docker version 20.10.14, build a224086349
Docker Compose version 2.3.4

**Comment by pinglin-zhang** [comment]
After tons of hours I finally figure it out how to solve this problem, which due to the system conflict in macOs. (So problem occurred and solved for macOS).

In my case, I got the error read as:
`response from daemon: Ports are not available: listen tcp 0.0.0.0:5000: bind: address already in use` on my docker container for COCO-Annotator.

This is because under the MacOS system, AirPlayer Server is also using the port :5000, which is in conflict with the localhost of COCO-Annotator.

To solve this:

**System Preferences › Sharing, and unchecking AirPlay Receiver to release port 5000** 

And then back to “cd coco-annotator” and “docker-compose up” on terminal. Up to this step I finally got to install COCO-Annotator and enter the localhost:5000.

If the problem is still not solved, try to do:

```
docker-compose down  # Stop container on current dir if there is a docker-compose.yml
docker rm -fv $(docker ps -aq)  # Remove all containers
sudo lsof -i -P -n | grep 5000  # List who's using the port 5000
kill -9 <process id> or kill -9 `sudo lsof -t -i:5000`  # remove other servers which using the port 5000
```



Good luck :)

**Comment by santosh-shriyan** [comment]
> Hello Everyone, I haven't worked with Docker before trying to use this tool. I followed all the installation steps from docker and COCO-Annotator both on Linux and Windows and it always gets stuck in the middle of the installation at the same place.
> 
> When I run `$ docker-compose up` I get this error: `.` `.` `.` `annotator_workers | [2021-10-12 00:17:02,694: INFO/MainProcess] mingle: all alone` `annotator_workers | [2021-10-12 00:17:02,807: INFO/MainProcess] celery@67c1eba9b51e ready.` `annotator_message_q | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> closing AMQP connection <0.778.0> (172.18.0.4:59266 -> 172.18.0.2:5672):` `annotator_message_q | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> missed heartbeats from client, timeout: 60s `
> 
> I would appreciate any kind of help. Am I doing something wrong?
> 
> Thank You

Did you solve it? Even I am stuck at this step.

I have also tried 
```
docker-compose down  # Stop container on current dir if there is a docker-compose.yml
docker rm -fv $(docker ps -aq)  # Remove all containers
sudo lsof -i -P -n | grep 5000  # List who's using the port 5000
kill -9 <process id> or kill -9 `sudo lsof -t -i:5000`  # remove other servers which using the port 5000
```
but still facing this problem.

**Comment by pinglin-zhang** [comment]
> > Hello Everyone, I haven't worked with Docker before trying to use this tool. I followed all the installation steps from docker and COCO-Annotator both on Linux and Windows and it always gets stuck in the middle of the installation at the same place.
> > When I run `$ docker-compose up` I get this error: `.` `.` `.` `annotator_workers | [2021-10-12 00:17:02,694: INFO/MainProcess] mingle: all alone` `annotator_workers | [2021-10-12 00:17:02,807: INFO/MainProcess] celery@67c1eba9b51e ready.` `annotator_message_q | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> closing AMQP connection <0.778.0> (172.18.0.4:59266 -> 172.18.0.2:5672):` `annotator_message_q | 2021-10-12 00:20:01.659862+00:00 [erro] <0.778.0> missed heartbeats from client, timeout: 60s `
> > I would appreciate any kind of help. Am I doing something wrong?
> > Thank You
> 
> Did you solve it? Even I am stuck at this step.
> 
> I have also tried
> 
> ```
> docker-compose down  # Stop container on current dir if there is a docker-compose.yml
> docker rm -fv $(docker ps -aq)  # Remove all containers
> sudo lsof -i -P -n | grep 5000  # List who's using the port 5000
> kill -9 <process id> or kill -9 `sudo lsof -t -i:5000`  # remove other servers which using the port 5000
> ```
> 
> but still facing this problem.

Yeah i solved it with this solution and I have shared it in my last reply (also see here https://github.com/jsbroks/coco-annotator/issues/517#issuecomment-1086626277). Good luck.

**Comment by LiuMingrui-Study** [comment]
I have encountered a similar problem, and the error message is as follows:
annotator_message_q | [error]<0.729.0>closing AMQP connection <0.729.0> (172.18.0.5:50396 -> 172.18.0.4:5672):
annotator_message_q | [error]<0.729.0> missed heartbeats from client, timeout: 60s 
But I checked the port occupation. Even if no program occupied port 5000, I still reported an error. Why？
